src/nile/core/compile.py

```diff
@@ -11,12 +11,14 @@
 )
 
 
-def compile(contracts, directory=None, account_contract=False):
+def compile(contracts, disable_hint_validation, directory=None, account_contract=False):
     """Compile cairo contracts to default output directory."""
     # to do: automatically support subdirectories
 
     contracts_directory = directory if directory else CONTRACTS_DIRECTORY
 
+    hint_validation = "--disable_hint_validation" if disable_hint_validation else ""
```

Comment:
I suggest we pass the flag as a boolean to `_compile_contract` and let that function handle the boolean -- that's the only function that should be aware of compiler flags.